proto/dcs.proto

```diff
@@ -25,17 +23,270 @@ message SetUserFlagRequest {
+  // Occurs when an aircraft is finished taking fuel.
+  message RefuelingStopEvent {
+    // he unit that was receiving fuel.
+    string initiator = 1;
+  }
+
+  // Occurs when any object is spawned into the mission.
+  message BirthEvent {
+    // The unit that was spawned.
+    string initiator = 1;
+  }
+
+  // Occurs when a system fails on a human controlled aircraft occurs.
+  message SystemFailureEvent {
+    // The unit the system failure occurred in.
+    string initiator = 1;
+  }
+
+  // Occurs when any aircraft starts its engines.
+  message EngineStartupEvent {
+    // The unit that starts its engines.
+    string initiator = 1;
+  }
+
+  message EngineShutdownEvent {
+    // Occurs when any aircraft shuts down its engines.
+    string initiator = 1;
+  }
+
+  // Occurs when a player takes direct control of a unit.
+  message PlayerEnterUnitEvent {
+    // The unit the player took control of.
+    string initiator = 1;
+  }
+
+  // Occurs when a player relieves direct control of a unit.
+  message PlayerLeaveUnitEvent {
+    // The unit the player relieves control of.
+    string initiator = 1;
+  }
+
+  // Occurs when a unit begins firing a machine gun- or autocannon-based weapon
+  // (weapons with a high rate of fire). Other weapons are handled by
+  // [ShotEvent].
+  message ShootingStartEvent {
+    // The unit that started firing.
+    string initiator = 1;
+  }
+
+  // Occurs when a unit stops firing a machine gun- or autocannon-based weapon.
+  // Event will always correspond with a [ShootingStartEvent] event.
+  message ShootingEndEvent {
+    // The unit that was shooting and has no stopped firing.
+    string initiator = 1;
+  }
+
+  // Occurs when marks get added to the mission by players or scripting
+  // functions.
+  message MarkAddEvent {
+    // The unit that added the mark.
+    string initiator = 1;
+    // The group the mark's visibility is restricted for.
+    uint64 group_id = 2;
+    // The coalition the mark's visibility is restricted for.
+    Coalition coalition = 3;
+    // The mark's id.
+    uint32 id = 4;
+    // The position the mark has been added at.
+    Position pos = 5;
+    // The mark's label.
+    string text = 6;
+  }
+
+  // Occurs when marks got changed.
+  message MarkChangeEvent {
+    // The unit that changed the mark.
+    string initiator = 1;
+    // The group the mark's visibility is restricted for.
+    uint64 group_id = 2;
+    // The coalition the mark's visibility is restricted for.
+    Coalition coalition = 3;
+    // The mark's id.
+    uint32 id = 4;
+    // The position of the changed mark.
+    Position pos = 5;
+    // The mark's label.
+    string text = 6;
+  }
+
+  // Occurs when marks get removed.
+  message MarkRemoveEvent {
+    // The unit that removed the mark.
+    string initiator = 1;
+    // The group the mark's visibility is restricted for.
+    uint64 group_id = 2;
+    // The coalition the mark's visibility is restricted for.
+    Coalition coalition = 3;
```

Comment:
`group_id` and `coalition` aren't always defined, how do we deal with that in protobuf / the RPC. Any ideas?

Reply:
I solved it by using `oneof`, feel free to let me know if you have a better idea.